### PR TITLE
feat: sidebar configuration depending on allowed roles provided in json file

### PR DESCRIPTION
**Pull request by dawidgil**

### CODE REVIEW

File: libs/ddhub-client-gateway-frontend/ui/utils/src/lib/config/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  },
+  auth: {
+    username: process.env['USERNAME'],
+    password: process.env['PASSWORD']
+  },
+  events: {
+    emitMode: asEnum(['SINGLE', 'BULK'], process.env['EVENTS_EMIT_MODE']) ?? defaults.eventsMode,
+    maxPerSecond: parseInt(process.env['EVENTS_MAX_PER_SECOND'] ?? defaults.eventsPerSecond, 10)
+  }
+}
```
Comment (driamend) [inline libs/ddhub-client-gateway-frontend/ui/utils/src/lib/config/index.ts:70]: not sure that we need this file at all, who is using it?

File: libs/ddhub-client-gateway-frontend/ui/login/src/lib/SetUserData.effects.ts [hunk 2]
```diff
@@ -5,20 +5,59 @@ import {
...
+  const roles = accountRoles
+    .filter(
+      (role) =>
+        role.status === RoleStatus.SYNCED &&
+        role.namespace.includes(config.iam.parentNamespace)
+    )
+    .map((role) => role.namespace);
+  const allowedRoutes = Object.keys(restrictions).map((key: string) => {
+    if (
+      restrictions[key].allowedRoles.some((allowedRole: string) =>
```
Comment (driamend) [inline libs/ddhub-client-gateway-frontend/ui/login/src/lib/SetUserData.effects.ts:41]: not sure that it's working, I was unable to restrict menu items

Comment (dawidgil) [reply libs/ddhub-client-gateway-frontend/ui/login/src/lib/SetUserData.effects.ts:41]: remove from mocks `topiccreator` role, and you shouldn't see topic management.

File: libs/ddhub-client-gateway-frontend/ui/utils/src/lib/config/index.ts (hunk 1 above)
Comment (dawidgil) [reply libs/ddhub-client-gateway-frontend/ui/utils/src/lib/config/index.ts:70]: it's old, just reused it but maybe i will just copy paste that part that i need.

File: libs/ddhub-client-gateway-frontend/ui/login/src/lib/SetUserData.effects.ts [hunk 3]
```diff
@@ -5,20 +5,61 @@ import {
...
+
+export const getRoutesToDisplay = (
+  accountRoles: Role[],
+  restrictions: RouteRestrictions
+): Set<string> => {
+  const roles = accountRoles
+    .filter(
+      (role) =>
+        role.status === RoleStatus.SYNCED &&
+        role.namespace.includes(process.env['PARENT_NAMESPACE'] ?? 'dsb.apps.energyweb.iam.ewc')
```
Comment (driamend) [inline libs/ddhub-client-gateway-frontend/ui/login/src/lib/SetUserData.effects.ts:35]: `process.env['PARENT_NAMESPACE']` will always be undefined, frontend can only see env with `NEXT_PUBLIC` prefix, also not sure what exactly is `dsb.apps.energyweb.iam.ewc`, who is using it?

File: libs/ddhub-client-gateway-frontend/ui/login/src/lib/CheckAccountOnInit.effects.ts [hunk 4]
```diff
@@ -6,17 +6,34 @@ import {
...
     setIsChecking(true);
-    queryClient
-      .fetchQuery(getIdentityControllerGetQueryKey(), identityControllerGet)
+    getIdentityData()
       .then((res) => {
-        setUserData(res as IdentityWithEnrolment);
+        setUserData(
+          res.identityData as IdentityWithEnrolment,
+          res.routeRestrictions
+        );
```
Comment (driamend) [inline libs/ddhub-client-gateway-frontend/ui/login/src/lib/CheckAccountOnInit.effects.ts:36]: still can be improved, we don't really need to rely on context for this data, both can be fetched with `react-query` and cached, to check loading status we can use something like `queryClient.isFetching({ queryKey: getIdentityControllerGetQueryKey() })`, but we can do it in another task or later

**Review by driamend — APPROVED** [review]

File: libs/ddhub-client-gateway-frontend/ui/login/src/lib/CheckAccountOnInit.effects.ts (hunk 4 above)
Comment (dawidgil) [reply libs/ddhub-client-gateway-frontend/ui/login/src/lib/CheckAccountOnInit.effects.ts:36]: It's working but i know it's not a good approach. Couldn't make it work as i want to with react query, i will try later to rewrite it.

**Pull request merged**